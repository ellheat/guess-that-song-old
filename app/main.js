/**
 * app.js
 *
 * This is the entry file for the application, only setup and boilerplate
 * code.
 */

import React from 'react';
import ReactDOM from 'react-dom';

// Needed for redux-saga es6 generator support
import 'babel-polyfill';

// Import all the third party stuff
import { Provider } from 'react-redux';
import createHistory from 'history/createBrowserHistory';
import { ConnectedRouter } from 'react-router-redux';
import injectTapEventPlugin from 'react-tap-event-plugin';
import FontFaceObserver from 'fontfaceobserver';
import 'normalize.css/normalize.css';
import configureStore from './modules/store';
import UnsupportedBrowserDetection from './utils/unsupportedBrowserDetection';

// Observe loading of Open Sans (to remove open sans, remove the <link> tag in
// the index.html file and this observer)
const openSansObserver = new FontFaceObserver('Open Sans', {});

// When Open Sans is loaded, add a font-family using Open Sans to the body
openSansObserver.load().then(() => {
  document.body.classList.add('fontLoaded');
}, () => {
  document.body.classList.remove('fontLoaded');
});

// Needed for onTouchTap
// Check this repo:
// https://github.com/zilverline/react-tap-event-plugin
injectTapEventPlugin();

// Create redux store with history
// this uses the singleton browserHistory provided by react-router
// Optionally, this could be changed to leverage a created history
// e.g. `const browserHistory = useRouterHistory(createBrowserHistory)();`
const initialState = {};
const browserHistory = createHistory();
const store = configureStore(initialState, browserHistory);

if (process.env.NODE_ENV === 'development') {
  if (!window.__REDUX_DEVTOOLS_EXTENSION__) {
    const DevToolsComponent = require('./utils/devtools.component').default;
    const devToolsRoot = window.document.createElement('div');

    window.document.body.appendChild(devToolsRoot);

    ReactDOM.render(
      <Provider store={store}>
        <DevToolsComponent />
      </Provider>,
      devToolsRoot
    );
  }
}

const render = () => {
  const NextApp = require('./routes').default;

  ReactDOM.render(
    <Provider store={store}>
      <ConnectedRouter history={browserHistory}>
        <NextApp />
      </ConnectedRouter>
    </Provider>,
    document.getElementById('app')
  );
};

const initApp = () => {
  const detection = new UnsupportedBrowserDetection();
  if (!detection.isSupported()) {
    return;
  }

  // Chunked polyfill for browsers without Intl support
  if (!window.Intl) {
    (new Promise((resolve) => {
      resolve(require('intl'));
    }))
      .then(() => Promise.all([
        require('intl/locale-data/jsonp/en.js'),
      ]))
      .then(() => render())
      .catch((err) => {
        throw err;
      });
  } else {
    render();
  }
};

initApp();

// Install ServiceWorker and AppCache in the end since
// it's not most important operation and if main code fails,
// we do not want it installed
if (process.env.NODE_ENV === 'production') {
  require('offline-plugin/runtime').install(); // eslint-disable-line global-require, import/no-extraneous-dependencies
}
