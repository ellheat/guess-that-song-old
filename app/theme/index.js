export { default as SpriteImage } from './spriteImage.component';

export * as sprites from './sprites';
