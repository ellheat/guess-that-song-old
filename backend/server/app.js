const serverless = require('serverless-http');
const express = require('express');
const path = require('path');
const port = require('../config/ports');
const app = express();
const router = express.Router(); // eslint-disable-line

const { getCharacters } = require('../config/characters');
const { getSpotifyToken, fetchPlaylist } = require('../utils/spotify');


const configureServer = () => new Promise((resolve) => {
  router.get('/', (req, res) => {
    res.send(express.static(path.join(__dirname, '../../build')));
  });
  app.listen(port.api, () => {
    console.log(`Backend listening on port ${port.api}!`.success); // eslint-disable-line
    getCharacters().then(() => console.log('Characters created'.success)); // eslint-disable-line
    getSpotifyToken().then(() => {
      console.log('Spotify token created'.success); // eslint-disable-line
      fetchPlaylist().then(() => resolve());
    });
  });
});


module.exports = {
  app,
  port,
  configureServer,
};

exports.handler = serverless(app);
