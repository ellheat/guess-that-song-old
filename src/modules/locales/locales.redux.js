import { Record } from 'immutable';
import { createReducer, createActions } from 'reduxsauce';

import { INITIAL_LOCALE } from '../../i18n';

export const { Types: LocalesTypes, Creators: LocalesActions } = createActions({
  setLanguage: ['language'],
}, { prefix: 'LOCALES_' });

export const LocalesRecord = new Record({
  language: INITIAL_LOCALE,
}, 'Locales');

export const INITIAL_STATE = new LocalesRecord({});

export const setLanguageHandler = (state = INITIAL_STATE, action) => state.set('language', action.language);
export const HANDLERS = {
  [LocalesTypes.SET_LANGUAGE]: setLanguageHandler,
};

export const reducer = createReducer(INITIAL_STATE, HANDLERS);

