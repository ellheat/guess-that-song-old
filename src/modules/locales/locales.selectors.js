import { createSelector } from 'reselect';
import { memoizeWith, identity } from 'ramda';
import { appLocales } from '../../i18n';

const getNextLanguage = memoizeWith(identity, (currentLanguage) => {
  return appLocales.find((language) => (currentLanguage !== language));
});

const selectLocalesDomain = state => state.locales;

export const selectLocalesLanguage = createSelector(
  selectLocalesDomain, state => state.get('language')
);

export const selectLocalesNextLanguage = createSelector(
  selectLocalesDomain, state => getNextLanguage(state.get('language'))
);
