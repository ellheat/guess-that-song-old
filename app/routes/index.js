import React, { Component } from 'react';
import { Route, Switch, Redirect, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { DEFAULT_LOCALE } from '../modules/locales/locales.redux';

import App from './app.container';
import { Home } from './home';
import { NotFound } from './notFound';

export class RootContainer extends Component {
  static propTypes = {
    location: PropTypes.object.isRequired,
  };

  render() {
    return (
      <Switch>
        <Route exact path="/" render={() => <Redirect to={DEFAULT_LOCALE} />} />


        <Route exact path="/404" component={NotFound} />

        <Route path="/:lang">
          <App location={this.props.location}>
            <Switch>
              <Route exact path="/:lang" component={Home} />

              <Route component={NotFound} />
            </Switch>
          </App>
        </Route>
      </Switch>
    );
  }
}

export default withRouter(RootContainer);
