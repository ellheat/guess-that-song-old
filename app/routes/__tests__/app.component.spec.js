import React from 'react';
import { expect } from 'chai';
import { shallow, mount } from 'enzyme';
import { spy } from 'sinon';

import { DEFAULT_LOCALE } from '../../modules/locales/locales.redux';
import { App } from '../app.component';


describe('App: Component', () => {
  const children = <div className="app__children">Children</div>;
  const defaultProps = {
    setLanguage: () => {
    },
    language: 'en',
    match: {
      params: {
        lang: 'en',
      },
    },
    location: {},
    history: {
      push: () => { },
    },
  };

  const component = (props) => (
    <App {...defaultProps} {...props}>
      {children}
    </App>
  );

  it('should not render App when language is not set', () => {
    const wrapper = shallow(component({ language: undefined }));
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render App when language is set', () => {
    const wrapper = shallow(component({ language: 'en' }));
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should redirect to /404 when given locale doesn\'t exist', () => {
    const router = {
      match: {
        params: {
          lang: 'non-existing-locale',
        },
      },
      history: {
        push: spy(),
      },
    };

    mount(component(router));
    expect(router.history.push).to.have.been.calledWith('/404');
  });

  it('should set DEFAULT_LOCALE when no lang param is given', () => {
    const setLanguage = spy();
    const router = {
      match: {
        params: {},
      },
      history: {
        push: () => {},
      },
    };

    mount(component({ ...router, setLanguage }));
    expect(setLanguage).to.have.been.calledWith(DEFAULT_LOCALE);
  });
});
