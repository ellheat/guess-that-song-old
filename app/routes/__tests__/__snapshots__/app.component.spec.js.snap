// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Component should not render App when language is not set 1`] = `""`;

exports[`App: Component should render App when language is set 1`] = `
<React.Fragment>
  <HelmetWrapper
    defaultTitle="Apptension React Boilerplate"
    defer={true}
    encodeSpecialCharacters={true}
    meta={
      Array [
        Object {
          "content": "Apptension's React Boilerplate application",
          "name": "description",
        },
      ]
    }
    titleTemplate="%s - Apptension React Boilerplate"
  />
  <IntlProvider
    locale="en"
    location={Object {}}
    messages={
      Object {
        "contact.title": "Contact page",
        "home.maintainerList.title": "Maintainers",
        "home.welcome": "Welcome to Apptension React Boilerplate!",
      }
    }
  >
    <div
      className="app__children"
    >
      Children
    </div>
  </IntlProvider>
</React.Fragment>
`;
